soda_machine.rb

```diff
@@ -1,18 +1,26 @@
 class SodaMachine
-  attr_reader :sodas, :cash
+  attr_reader  :cash, :sodas, :sold_soda
 
   def initialize(args = {})
     @sodas = args[:sodas]
     @cash = args[:cash]
+    @sold_soda = nil
   end
 
   def current_inventory_count
+    @sodas.length
   end
 
   def find_soda(soda_brand)
+    if @sodas.include? soda_brand
+      soda_brand
+    else
+      nil
+    end
   end
 
   def sell(soda_brand)
+    find_soda(soda_brand)
```

Comment:
Good start here, using your previously written method to avoid recreating work you've already done.